### PR TITLE
Adds traitor steal objective: the Supermatter Sliver. (Fixed)

### PR DESCRIPTION
**Pull request by harmonyn**
#:cl: bandit
add: The Syndicate has been observed with supermatter technology. Nanotrasen is investigating potential tampering with its supermatter devices.
add: Traitors can steal supermatter slivers now. It works much like the plutonium core.
/:cl:

New PR, changes requested by @Cyberboss made, merge conflicts resolved, test issues resolved

Traitors have a new steal objective, to extract a sliver of a supermatter crystal. This works much like stealing the plutonium core:

- Use a special scalpel (different fluff is welcome) to slice a sliver off. Like the plutonium core, this sliver will periodically emit radiation if not stored.
- Pick the sliver up with tongs, also provided. If you or anything else touches the crystal, then it dusts. If you use the tongs to attack a person, the resulting force will dust you as well (so you can't go around instakilling people). The sliver will also be destroyed, to prevent a chain reaction of people dusting people.
- Use the tongs to put it in the container, provided as well.
- Once you extract a sliver, the supermatter will gain energy, requiring you to have protection. The exact amount of energy can be tweaked, the idea is that delamination will not happen immediately but that it is serious, noticeable sabotage and the traitor should probably get out fast.
- Both the scalpel and tongs do burn instead of brute damage, because they are made of SEARING COLD FREON THIS IS GREAT FLUFF sigh

What this adds to the game: Steal objectives should be more difficult than they are. No one gives a shit about, say, the blueprints. The supermatter also lacks a meaningful traitor interaction that is not just releasing a singularity. This provides that. 

Other changes:

- In order to prevent the sliver from being dragged added a can_be_pulled proc to movable atoms, and moved some of the start_pulling checks into that proc.

### CODE REVIEW

**Comment by bgobandit** [comment]
WHY IS THE DME DOING THIS SHIT

**Comment by loqoman** [comment]
I dig this. New traitor interaction with the new engine.

+1

**Comment by RandomMarine** [comment]
The container should be disguised as a super matter bin.

**Comment by octareenroon91** [comment]
I guess that's a decent pun.

File: code/game/gamemodes/objective_items.dm [hunk 1]
```diff
@@ -84,6 +84,14 @@
 /datum/objective_item/steal/nuke_core/New()
 	special_equipment += /obj/item/weapon/storage/box/syndie_kit/nuke
 
+/datum/objective_item/steal/supermatter
+	name = "a sliver of a supermatter crystal. Be sure to use the proper safety equipment when extracting the sliver!"
+	targetitem = /obj/item/nuke_core/supermatter_sliver
+	difficulty = 15
+
+/datum/objective_item/steal/supermatter/New()
+	special_equipment += /obj/item/weapon/storage/box/syndie_kit/supermatter
```
Comment (Cyberboss) [inline code/game/gamemodes/objective_items.dm]: Call ..()

File: code/game/objects/items/theft_tools.dm [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+/obj/item/nuke_core_container
+	name = "nuke core container"
+	desc = "Solid container for radioactive objects."
+	icon = 'icons/obj/nuke_tools.dmi'
+	icon_state = "core_container_empty"
+	item_state = "tile"
+	var/obj/item/nuke_core/core
+
+/obj/item/nuke_core_container/Destroy()
+	QDEL_NULL(core)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: return ..()

File: code/game/objects/items/theft_tools.dm [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+	icon_state = "plutonium_core"
+	item_state = "plutoniumcore"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	var/pulse = 0
+	var/cooldown = 0
+	var/pulseicon = "plutonium_core_pulse"
+
+/obj/item/nuke_core/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: STOP_PROCESSING in Destroy

File: code/game/objects/items/theft_tools.dm [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return TRUE
+
+/obj/item/nuke_core_container/proc/seal()
+	if(istype(core))
+		STOP_PROCESSING(SSobj, core)
+		icon_state = "core_container_sealed"
+		playsound(src, 'sound/items/Deconstruct.ogg', 60, 1)
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[src] is permanently sealed, [core]'s radiation is contained.</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: \The [src]

\the [core]

File: code/game/objects/items/theft_tools.dm [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+/obj/item/nuke_core_container/attackby(obj/item/nuke_core/core, mob/user)
+	if(istype(core))
+		if(!user.temporarilyRemoveItemFromInventory(core))
+			to_chat(user, "<span class='warning'>The [core] is stuck to your hand!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: \The

File: code/game/objects/items/theft_tools.dm [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+/obj/item/nuke_core/supermatter_sliver/attackby(obj/item/W, mob/living/user, params)
+	if(istype(W, /obj/item/weapon/hemostat/supermatter))
+		var/obj/item/weapon/hemostat/supermatter/tongs = W
+		if (tongs.sliver)
+			to_chat(user, "<span class='notice'>\The [tongs] is already holding a supermatter sliver!</span>")
+			return FALSE
+		forceMove(tongs)
+		tongs.sliver = src
+		tongs.icon_state = "supermatter_tongs_loaded"
+		to_chat(user, "<span class='notice'>You carefully pick up [src] with [tongs].</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: \the [src]

\the [tongs]

File: code/game/objects/items/theft_tools.dm [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+	playsound(get_turf(user), 'sound/effects/supermatter.ogg', 50, 1)
+	ded.dust()
+
+/obj/item/nuke_core_container/supermatter
+	name = "supermatter receptacle"
+	desc = "A tiny receptacle that releases an inert freon mix upon sealing, allowing a sliver of a supermatter crystal to be safely stored.."
+	var/obj/item/nuke_core/supermatter_sliver/sliver
+
+/obj/item/nuke_core_container/supermatter/Destroy()
+	QDEL_NULL(sliver)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: return ..()

File: code/game/objects/items/theft_tools.dm [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return TRUE
+
+/obj/item/nuke_core_container/supermatter/seal()
+	if(istype(sliver))
+		STOP_PROCESSING(SSobj, sliver)
+		icon_state = "core_container_sealed"
+		playsound(loc, 'sound/items/Deconstruct.ogg', 60, 1)
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[src] is permanently sealed, [sliver] is safely contained.</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: \The [src]

\the [sliver]

File: code/game/objects/items/theft_tools.dm [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+	icon_state = "core_container_loaded"
+	to_chat(user, "<span class='warning'>Container is sealing...</span>")
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return TRUE
+
+/obj/item/nuke_core_container/supermatter/seal()
+	if(istype(sliver))
+		STOP_PROCESSING(SSobj, sliver)
+		icon_state = "core_container_sealed"
+		playsound(loc, 'sound/items/Deconstruct.ogg', 60, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: playsound(src, ...

File: code/game/objects/items/theft_tools.dm [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+		forceMove(tongs)
+		tongs.sliver = src
+		tongs.icon_state = "supermatter_tongs_loaded"
+		to_chat(user, "<span class='notice'>You carefully pick up [src] with [tongs].</span>")
+	else if(istype(W, /obj/item/weapon/scalpel/supermatter) || istype(W, /obj/item/nuke_core_container/supermatter/)) // we don't want it to dust
+		return
+	else
+		to_chat(user, "<span class='notice'>As it touches \the [src], both \the [src] and \the [W] burst into dust!</span>")
+		radiation_pulse(get_turf(user), 1, 2, 10, 1)
+		playsound(get_turf(user), 'sound/effects/supermatter.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: playsound(src, ...

File: code/game/objects/items/theft_tools.dm [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+	name = "supermatter extraction tongs"
+	desc = "A pair of tongs made from condensed freon gas, searingly cold to the touch, that can safely grip a supermatter sliver."
+	icon = 'icons/obj/nuke_tools.dmi'
+	icon_state = "supermatter_tongs"
+	toolspeed = 0.75
+	damtype = "fire"
+	var/obj/item/nuke_core/supermatter_sliver/sliver
+
+/obj/item/weapon/hemostat/supermatter/Destroy()
+	QDEL_NULL(sliver)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: return ..()

File: code/game/objects/items/theft_tools.dm [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+		message_admins("[src] has consumed [key_name_admin(victim)] [ADMIN_JMP(src)].")
+		investigate_log("has consumed [key_name(victim)].", "supermatter")
+	else
+		investigate_log("has consumed [AM].", "supermatter")
+		qdel(AM)
+	user.visible_message("<span class='danger'>As [user] touches \the [AM] with \a [src], silence fills the room...</span>",\
+			"<span class='userdanger'>You touch \the [AM] with \the [src], and everything suddenly goes silent.</span>\n<span class='notice'>\The [AM] flashes into dust, and soon as you can register this, you do as well.</span>",\
+			"<span class='italics'>Everything suddenly goes silent.</span>")
+	radiation_pulse(get_turf(user), 2, 4, 50, 1)
+	playsound(get_turf(user), 'sound/effects/supermatter.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: playsound(src, ...

File: code/modules/power/supermatter/supermatter.dm [hunk 13]
```diff
@@ -463,7 +463,14 @@
 /obj/machinery/power/supermatter_shard/attackby(obj/item/W, mob/living/user, params)
 	if(!istype(W) || (W.flags & ABSTRACT) || !istype(user))
 		return
-	if(user.drop_item(W))
+	if(istype(W, /obj/item/weapon/scalpel/supermatter))
+		playsound(src, W.usesound, 100, 1)
+		to_chat(user, "<span class='notice'>You carefully begin to scrape \the [src] with \the [W]...</span>")
+		if(do_after(user, 60*W.toolspeed,target=src))
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: `do_after(user, 60 * W.toolspeed, TRUE, src)`

File: tgstation.dme [hunk 14]
```diff
@@ -174,7 +174,6 @@
 #include "code\controllers\subsystem\inbounds.dm"
 #include "code\controllers\subsystem\ipintel.dm"
 #include "code\controllers\subsystem\job.dm"
-#include "code\controllers\subsystem\language.dm"
```
Comment (Cyberboss) [inline tgstation.dme]: You broke the fuck out of the dme

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/theft_tools.dm (hunk 4 above)
Comment (RemieRichards) [inline code/game/objects/items/theft_tools.dm]: no. it's automatic we've told you thiiiis.

File: code/game/objects/items/theft_tools.dm (hunk 5 above)
Comment (RemieRichards) [inline code/game/objects/items/theft_tools.dm]: ^ see above cyber

File: code/game/objects/items/theft_tools.dm (hunk 6 above)
Comment (RemieRichards) [inline code/game/objects/items/theft_tools.dm]: ^ see above cyber

File: code/game/objects/items/theft_tools.dm (hunk 8 above)
Comment (RemieRichards) [inline code/game/objects/items/theft_tools.dm]: ^ see above cyber

**Comment by lzimann** [comment]
Know you are busy and all of that, but cmon

**Comment by bgobandit** [comment]
don't close this I'm trying to fix the dme issue

**Comment by lzimann** [comment]
Just revert the DME to match the one in upstream, then you just tick your file again.

**Comment by DoctoreMad** [comment]
Guys lets have an traitor objective that requires you to destabilize the station's power generator for the meme

**Comment by octareenroon91** [comment]
Self-wrecking engine is just as much a return to form.

The issue of actual risk has already been discussed too.

File: code/game/objects/items/theft_tools.dm [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+
+/obj/item/weapon/paper/nuke_instructions
+	info = "How to break into a Nanotrasen self-destruct terminal and remove its plutonium core:<br>\
+	<ul>\
+	<li>Use a screwdriver with a very thin tip (provided) to unscrew the terminal's front panel</li>\
+	<li>Dislodge and remove the front panel with a crowbar</li>\
+	<li>Cut the inner metal plate with a welding tool</li>\
+	<li>Pry off the inner plate with a crowbar to expose the radioactive core</li>\
+	<li>Use the core container to remove the plutonium core; the container will take some time to seal</li>\
+	<li>???</li>"
```
Comment (lzimann) [inline code/game/objects/items/theft_tools.dm]: Missing a `</ul>` tag

File: code/game/objects/items/theft_tools.dm [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+/obj/item/weapon/paper/supermatter_sliver_instructions
+	info = "How to safely extract a supermatter sliver:<br>\
+	<ul>\
+	<li>Approach an active supermatter crystal with proper protective gear. DO NOT MAKE PHYSICAL CONTACT.</li>\
+	<li>Use a supermatter scalpel (provided) to slice off a sliver of the crystal.</li>\
+	<li>Use supermatter extraction tongs (also provided) to safely remove the sliver.</li>\
+	<li>Physical contact of any object with the sliver will dust the object, as well as yourself.</li>\
+	<li>Use the tongs to place the sliver into the provided container, which will take some time to seal</li>\
+	<li>Get the hell out before the crystal delaminates.</li>\
+	<li>???</li>"
```
Comment (lzimann) [inline code/game/objects/items/theft_tools.dm]: Same here.

**Comment by bgobandit** [comment]
all comments upthread should be addressed idk why they still show up

File: code/game/atoms_movable.dm [hunk 17]
```diff
@@ -684,3 +684,14 @@
 //Returns an atom's power cell, if it has one. Overload for individual items.
 /atom/movable/proc/get_cell()
 	return
+
+/atom/movable/proc/can_be_pulled(user)
+	if(src == user || !isturf(loc))
+		return FALSE
+	if(anchored || throwing)
+		return FALSE
+	if(isliving(src))
```
Comment (AnturK) [inline code/game/atoms_movable.dm]: No no, move this to living level /can_be_pulled

**Comment by AnturK** [comment]
@bgobandit  Merge conflicts.

**Comment by lzimann** [comment]
@bgobandit

**Comment by bgobandit** [comment]
@AnturK I literally have no idea what the fuck you are asking, the entire point of moving it was so it would be under the object, not mob/living

**Review by Cyberboss — CHANGES_REQUESTED** [review]
You deleted the tgstation-server submodule

File: code/game/atoms_movable.dm [hunk 18]
```diff
@@ -686,3 +686,14 @@
...
+
+/atom/movable/proc/can_be_pulled(user)
+	if(src == user || !isturf(loc))
+		return FALSE
+	if(anchored || throwing)
+		return FALSE
+	if(isliving(src))
+		var/mob/living/L = src
+		if(L.buckled && L.buckled.buckle_prevents_pull)
+			return FALSE
```
Comment (Cyberboss) [inline code/game/atoms_movable.dm]: Do what @AnturK said: Remove these 4 lines. Create `/mob/living/can_be_pulled()` in `code/modules/mob/living/living.dm` and fill it out like this:

```
/mob/living/can_be_pulled()
    return ..() && !(buckled && buckled.buckle_prevents_pull)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**